### PR TITLE
UI Testing For Demo Apps

### PR DESCRIPTION
**Pull request by twnolan**
## Motivation
Automated UI testing is required to catch any bugs in the demo applications

## Description
Updating old UI tests for raincatcher-1.0 and implement functionality that was previously unimplemented

## Progress Completed So Far
- [x] Implement page objects for mobile
- [x] Implement services for mobile
- [x] Implement tests for mobile
- [x] Add data from mobile to data objects
- [x] Check old content for grammar and spelling mistakes
- [x] Update some old locators and selectors
- [x] Prepare local conf file
- [x] Remove files not required
- [x] Replace all `it()` with `step()`
- [x] README.md file
- [x] Update portal auth page object to match new UI
- [x] added portal service object and portal tests
- [x] updated portal tests to use new service object

### CODE REVIEW

**Comment by twnolan** [comment]
@wtrocki here is the WIP PR

File: tests/e2e-ui-tests/data/workorders.do.js
Comment (wtrocki) [inline tests/e2e-ui-tests/data/workorders.do.js:21]: This is no longer used as start time should be setup on mobile, same with finish time. 
This fields are not needed.

File: tests/e2e-ui-tests/data/workorders.do.js
Comment (wtrocki) [inline tests/e2e-ui-tests/data/workorders.do.js:19]: We removed this data from top level. 
Everything is ok here but it's best to merge that now as when we will make changes we will also adjust this tests. 

ping @paolobueno

File: tests/e2e-ui-tests/page-objects/mobile/workflow/main.po.js
Comment (wtrocki) [inline tests/e2e-ui-tests/page-objects/mobile/workflow/main.po.js:1]: I think that we may just load some sample steps for purposes of this tests.
Vehicle inspection step it's not the best one for this purposes as it has some custom elements. 
From our point of view this elements doesn't matter at all as Vehicle inspection step is done only for sales demo purposes. We may just have couple different steps to make sure that we support branching etc.

With that in mind I will not invest more time on this particular test. Just FYI

File: tests/e2e-ui-tests/page-objects/portal/auth/login.po.js
Comment (wtrocki) [inline tests/e2e-ui-tests/page-objects/portal/auth/login.po.js:3]: We are lucky enough that this will work for both passport and keycloak!

File: tests/e2e-ui-tests/page-objects/portal/navigation.po.js
Comment (wtrocki) [inline tests/e2e-ui-tests/page-objects/portal/navigation.po.js]: Looks like we have a lot of legacy categories there. Remove that

File: tests/e2e-ui-tests/page-objects/portal/workorder/new.po.js
Comment (wtrocki) [inline tests/e2e-ui-tests/page-objects/portal/workorder/new.po.js:34]: No longer available - let's merge it this way and we can adjust that later. Just letting you know.

**Review by wtrocki — DISMISSED** [review]
Exceptional work! I'm ok to merge this after we will get all your points adressed. I do not think we should address all the problems and entire team can really help to maintain this once it's merged. You have build a really solid base. Thanks for that!

File: tests/e2e-ui-tests/data/workorders.do.js
Comment (paolobueno) [reply tests/e2e-ui-tests/data/workorders.do.js:19]: +1, take a look at the TS wfm module to confirm which fields we still have:

https://github.com/feedhenry-raincatcher/raincatcher-core/blob/master/client/wfm/src/workorder/WorkOrder.ts#L10

I think this is a nice opportunity to take a look at how we'd go ahead and porting these tests to work with angular2, which seems to use protractor as well.

File: tests/e2e-ui-tests/.eslintrc
Comment (paolobueno) [inline tests/e2e-ui-tests/.eslintrc]: We should avoid having independent eslint files into the modules, let's try to inherit from the root and add the extra variables these files need (i.e. `env.protractor`, globals, relevant rules, etc):

[root eslintrc](https://github.com/feedhenry-raincatcher/raincatcher-angularjs/blob/master/package.json#L7)
[eslint docs](https://eslint.org/docs/user-guide/configuring#using-a-configuration-file)

Comment (paolobueno) [inline tests/e2e-ui-tests/.eslintrc:43]: There seem to be more rules to this file than the one we're using in the other modules, do you see any of them being interesting to the rest of the repo?

Eventually I'd like to bubble the eslint config files to the root of the repo and at most extend them from each module, see related [eslint docs](https://eslint.org/docs/user-guide/configuring#using-a-configuration-file)

**Comment by wtrocki** [comment]
@twnolan - let's add some minimal readme and resolve some problems mentioned above and we can merge that. Follow up ticket to be created.

**Comment by wtrocki** [comment]
@twnolan - Tests here give us solid base, but currently our flow changed (we have 1 new step and some major changes are required). My proposition is to merge this change as they aren't breaking any causing any harm in master and build set of tickets to tackle this work that may overspill after GA release. It will be easier to maintain and test that as well when having this changes on master.

**Review by wtrocki — APPROVED** [review]

**Comment by twnolan** [comment]
@wtrocki I agree, the flow change will cause some test logic to be rewritten. We can improve on these once we have the main work on demo apps completed

**Pull request merged**
